python/lsst/daf/butler/cli/cmd/commands.py

```diff
@@ -159,12 +161,61 @@ def query_collections(*args, **kwargs):
 @options_file_option()
 def query_dataset_types(*args, **kwargs):
     """Get the dataset types in a repository."""
-    print(yaml.dump(cli_handle_exception(queryDatasetTypes, *args, **kwargs), sort_keys=False))
+    print(yaml.dump(cli_handle_exception(script.queryDatasetTypes, *args, **kwargs), sort_keys=False))
 
 
 @click.command()
 @repo_argument(required=True)
 @click.argument('dataset-type-name', nargs=1)
 def remove_dataset_type(*args, **kwargs):
-    """Remove a dataset type definition from a reopsitory."""
-    cli_handle_exception(removeDatasetType, *args, **kwargs)
+    """Remove a dataset type definition from a repository."""
+    cli_handle_exception(script.removeDatasetType, *args, **kwargs)
+
+
+@click.command()
+@repo_argument(required=True)
+@glob_argument(help="GLOB is one or more glob-style expressions that fully or partially identify the "
+                    "dataset types to be queried.")
+@click.option("--collections",
```

Comment:
we have a shared option "--collection-type" used in other places. What do you think about using that here?

Reply:
That's a different kind of argument entirely, actually; `--collections` here is more like the positional `COLLECTIONS` argument to `butler query-collections`.  FWIW, `queryDatasets` does internally expand a `collections` argument that includes one or more patterns into an explicit list of collections in much the same way as `queryCollections`, and that _could_ accept collection types from the user to provide more fine-grained control over what's in that expanded list.  But that'd first require a change to `Registry.queryDatasets` itself, and hence is probably out of scope for this ticket.